src/popup/components/TransactionList.vue

```diff
@@ -73,133 +63,150 @@ export default {
-              ? tr.tx?.contractId === this.token
-              : (!tr.tx.contractId
-                || !isFungibleTokenTx(tr)))))
-          .filter((tr) => {
-            switch (this.displayMode.filter) {
-              case 'all':
-                return true;
-              case 'dex':
-                return this.getDexContracts && tr.tx.contractId && (
-                  this.getDexContracts.router.includes(tr.tx.contractId)
-                  || this.getDexContracts.wae?.includes(tr.tx.contractId)
-                );
-              case 'out':
-                return (this.compareCaseInsensitive(tr.tx.type, SCHEMA.TX_TYPE.spend)
-                    && tr.tx.senderId === address)
-                  || (isFungibleTokenTx(tr)
-                    && this.compareCaseInsensitive(tr.tx.type, SCHEMA.TX_TYPE.contractCall)
-                    && tr.tx.callerId === address);
-              case 'in':
-                return (this.compareCaseInsensitive(tr.tx.type, SCHEMA.TX_TYPE.spend)
-                    && (tr.tx.recipientId === address || (tr.tx.senderId !== address && tr.tx.recipientId.startsWith('nm_'))))
-                  || (isFungibleTokenTx(tr)
-                    && this.compareCaseInsensitive(tr.tx.type, SCHEMA.TX_TYPE.contractCall)
-                    && tr.recipient === address);
-              case 'tips':
-                return (tr.tx.contractId
-                  && (activeNetwork.tipContractV1 === tr.tx.contractId
-                    || activeNetwork.tipContractV2 === tr.tx.contractId)
-                  && (tr.tx.function === 'tip' || tr.tx.function === 'retip')) || tr.claim;
-              default:
-                throw new Error(`Unknown display mode type: ${this.displayMode.filter}`);
-            }
-          })
-          .filter(
-            (tr) => !this.searchTerm
-              || this.getTxSymbol(tr)
-                .toLocaleLowerCase()
-                .includes(this.searchTerm.toLocaleLowerCase()),
-          )
-          .sort((a, b) => {
-            const arr = [a, b].map((e) => new Date(e.microTime));
-            if (this.displayMode.rotated) arr.reverse();
-            return (arr[0] - arr[1]) || a.pending;
-          })
-          .slice(0, this.maxLength || Infinity);
-      },
-    }),
-    showSearchAndFilters() {
-      return (
-        this.showFilters
-        || this.displayMode.filter !== 'all'
-        || this.searchTerm
-      );
-    },
-  },
-  mounted() {
-    this.loadMore();
-    const polling = setInterval(() => this.getLatest(), 10000);
-    const appInner = document.querySelector('.app-inner');
-
-    appInner.addEventListener('scroll', this.checkLoadMore);
-    window.addEventListener('scroll', this.checkLoadMore);
-    this.$once('hook:destroyed', () => {
-      appInner.removeEventListener('scroll', this.checkLoadMore);
-      window.removeEventListener('scroll', this.checkLoadMore);
-      clearInterval(polling);
-      this.isDestroyed = true;
+  setup(props, { root }) {
```

Comment:
Thing to consider here - what do you think of moving all of the things related to transactions to composable?

Reply:
in others step we have a lot PR with changes here, i think we must refactor this after all changes

Reply:
https://aeternity-blockchain.atlassian.net/browse/SW-533